model/notification/center/notification_center.go

```diff
@@ -288,9 +297,48 @@ func sendMail(inst *instance.Instance, p *notification.Properties, n *notificati
 	if err != nil {
 		return err
 	}
-	_, err = job.System().PushJob(inst, &job.JobRequest{
-		WorkerType: "sendmail",
-		Message:    msg,
-	})
-	return err
+	return pushJobOrTrigger(inst, msg, "sendmail", at)
+}
+
+func pushJobOrTrigger(inst *instance.Instance, msg job.Message, worker, at string) error {
+	if at == "" {
+		_, err := job.System().PushJob(inst, &job.JobRequest{
+			WorkerType: worker,
+			Message:    msg,
+		})
+		return err
+	}
+	t, err := job.NewTrigger(inst, job.TriggerInfos{
+		Type:       "@at",
+		WorkerType: worker,
+		Arguments:  at,
+	}, msg)
+	if err != nil {
+		return err
+	}
+	return job.System().AddTrigger(t)
+}
+
+// Make sure the stack always fallback on mail when the other channels has
+// failed.
+func appendMailToChannels(channels []string) []string {
```

Comment:
`ensureAtLeastMail` ?
`ensureFallbackMail` ?

This would remove the need for a docstring and would be clearer at the callpoint IMHO.